src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -587,6 +600,67 @@ int sd_dhcp6_lease_get_captive_portal(sd_dhcp6_lease *lease, const char **ret) {
         return 0;
 }
 
+int dhcp6_lease_add_vendor_option(sd_dhcp6_lease *lease, const uint8_t *optval, size_t optlen, size_t base_offset) {
+        int r;
+        uint16_t enterprise_id;
+
+        assert(lease);
+        assert(optval || optlen == 0);
+
+        lease->vendor_options = set_new(NULL);
+        enterprise_id = unaligned_read_be16(optval + offsetof(DHCP6Option, len));
```

Comment:
Please check `optlen` before read. Otherwise, networkd may be segfault by malformed messages.

Reply:
Fixed.